### ISSUE TITLE
JS libsodium won't initalize

### ISSUE THREAD

**Issue by luca992** [ionspin/kotlin-multiplatform-libsodium#23]
On JS I can't get initialize() to work. It fails with this error:
```
ReferenceError: $module$libsodium_wrappers_sumo is not defined
	at 0.doResume_5yljmg(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:294177)
	at JsSodiumLoader.load_e5ock0(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:294230)
	at 1.doResume_5yljmg(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:294370)
	at LibsodiumInitializer.initialize_slkfnq(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:294405)
	at slambda.doResume_5yljmg(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:303806)
	at slambda.invoke_nx893v(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:303792)
	at slambda.$testBody(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:303861)
	at slambda.doResume_5yljmg(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:302120)
	at slambda.invoke_c8cfpt(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:302106)
	at <global>.l(/var/folders/q4/bk4942ys3g5_gsbchw20c6k00000gn/T/_karma_webpack_796110/commons.js:302158)
```
Any idea what is going on? (this is in a test function, if that matters) Maybe it has to do with webpack5?

**Comment by ionspin** [comment]
Hi @luca992 , libsodium sumo is the js release of the libsodium library (sumo being the one with more primitives available), and without seeing exactly how you are packaging the final artifacts I don't think I can help, although I'm not an JavaScript expert so even with seeing the way you build the final artifacts I might not be able to help. 

You might be right it has to do with webpack 5, I have the following line in my projects `gradle.properties` to set the webpack to 4: 
``` kotlin.js.webpack.major.version=4 ```

See if that helps. 

If that solves the issue I'll add to documentation that webpack 4 is required, and I'll see if I can test with webpack 5 when I have some free time.

**Comment by luca992** [comment]
Didn't work.

But, I think the problem is how you get an instance of JsSodiumInterface?
I think JsSodiumInterface should be an external class annotated with `@JsModule`: https://kotlinlang.org/docs/js-modules.html#jsmodule-annotation

is there any reason you get an instance with `js("\$module\$libsodium_wrappers_sumo")`?

**Comment by ionspin** [comment]
> is there any reason you get an instance with js("\$module\$libsodium_wrappers_sumo"), I don't think that is standard practice?

When I was writing this code intially `@JsModule` didn't work and would actually fail with the undefined error.

I had to jump through a lot of hoops to get the js to work at all, the kotlin js compiler was unstable (at the time) and would fail in interesting and creative ways that it required a bunch of workarounds. I currently don't have time to experiment against latest kotlin js compiler, and in multiplatform projects that I am using the library in it works fine (for JS in my case that is browser flavor, I don't have node projects)

With that said, I'd be very happy to accept a pull request that improves this in any way!

**Comment by luca992** [comment]
Well, fixed it. 👍 #24

**Comment by luca992** [comment]
Not really sure how it works at all in it's current state in your project haha. Calling ready just didn't work at all for me before the module fix I made

**Comment by ionspin** [comment]
Awesome, thank you very much for the contribution, I'll give it a look, run the tests and get back to you.

**Comment by ionspin** [comment]
#24 is merged, CI/CD is done and the fix is available in 0.8.6-SNAPSHOT version. Thanks for the contribution that fixed the issue!
